### PR TITLE
Announce function key shortcut keys in java swing applications

### LINKED ISSUE #10994: Keyboard shortcut is sometimes not properly read for Java applications
In a Java application I have a menu item with Ctrl-F6 set as accelerator.
NVDA reads the shortcut as "control plus" without reading the F6 key at all. JAWS properly reads the entire shortcut.

      public static void main(String[] args) {
        JFrame fr = new JFrame();
        JMenuBar mb = new JMenuBar();
        JMenu menu = new JMenu("windows");
        JMenuItem mi = new JMenuItem("abc");
        mi.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_F6, KeyEvent.CTRL_DOWN_MASK));
        menu.add(mi);
        mb.add(menu);
        fr.setJMenuBar(mb);
        fr.setSize(500, 500);
        fr.setVisible(true);
      }

### PR DESCRIPTION
**Pull request by mwhapples**
### Link to issue number:
Closes #13643 
### Summary of the issue:
In a Java swing application NVDA was not announcing function keys in the shortcut keys. I also suspect it was not announcing other special keys like backspace, delete, cursor keys, etc.
### Description of how this pull request fixes the issue:
java access bridge defines modifier bits for function keys and control code keys and then uses specific values in the character field. This pull request checks for these bits and handles the shortcut key accordingly.
### Testing strategy:
Manual and unit tests.
### Known issues with pull request:
There are some keys not defined in the java access bridge constants (eg. space) and so those will be handled like normal character keys, I am not sure whether this is going to lead to correct speaking of space and similar keys. Also there are some keys which java supports but NVDA does not, these will be ignored by NVDA, as it used to.
### Change log entries:
New features
Changes
Bug fixes
Fix issue where function key shortcuts were not spoken in Java swing applications.
For Developers

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests Done
  - System (end to end) tests Not done, only small amount of code which could be covered by this anyway.
  - Manual testing Done
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by mwhapples** [comment]
I am aware of the lint failure, but I could not find a code style which would satisfy it. The issue seems to be about indentation and when I fix the one it leads to another which implies doing it the way I originally did it. So I have done it the closest to the guidelines in the contributor documentation.

**Comment by josephsl** [comment]
Hi, this is one of the reasons why I tend to perform a general lint on surrounding code (mostly at the function/method level). I also use a customized Flake8 config file used for my own add-ons wen linting NVDA Core code from scratch (the obvious advantage is that I can look at what Flake8 says about the entire file and lint the modified parts). Thanks.

**Comment by mwhapples** [comment]
I have taken a look about things like keys such as escape, enter, space, tab, etc. NVDA just reads the first letter of the words (eg. S for space, T for tab, etc). The biggest problem is that enter and escape are both E. I also tested that with Jaws and Jaws does similar for these shortcut keys, so I think it is a Java Access Bridge limitation/bug, so this PR is probably the best we can do at the moment.

**Comment by mwhapples** [comment]
This also fixes issue #10994

File: source/JABHandler.py [hunk 1]
```diff
@@ -226,14 +227,60 @@ class AccessibleTableCellInfo(Structure):
...
-ACCESSIBLE_ALT_KEYSTROKE=8
-ACCESSIBLE_ALT_GRAPH_KEYSTROKE=16
-ACCESSIBLE_BUTTON1_KEYSTROKE=32
-ACCESSIBLE_BUTTON2_KEYSTROKE=64
-ACCESSIBLE_BUTTON3_KEYSTROKE=128
+
+
+class Accessible(IntFlag):
+	"""
+	TODO: The documentation for this mapping is here:
```
Comment (seanbudd) [inline source/JABHandler.py:234]: @mwhapples would you be able to add to this docstring

File: source/JABHandler.py [hunk 2]
```diff
@@ -226,14 +227,60 @@ class AccessibleTableCellInfo(Structure):
...
+ACCESSIBLE_ALT_GRAPH_KEYSTROKE = Accessible.ALT_GRAPH_KEYSTROKE
+ACCESSIBLE_BUTTON1_KEYSTROKE = Accessible.BUTTON1_KEYSTROKE
+ACCESSIBLE_BUTTON2_KEYSTROKE = Accessible.BUTTON2_KEYSTROKE
+ACCESSIBLE_BUTTON3_KEYSTROKE = Accessible.BUTTON3_KEYSTROKE
+
+
+class AccessibleVK(IntEnum):
+	"""
+	The supported control code keys related to Accessible.CONTROLCODE_KEYSTROKE.
+	TODO: The documentation for this mapping is here:
```
Comment (seanbudd) [inline source/JABHandler.py:266]: @mwhapples would you be able to add to this docstring

**Review by seanbudd — COMMENTED** [review]
I've pushed a commit to fix the lint.

I've also pushed a commit to change the constants to enums. 
This is preferred for new groups of constants, however we need to keep the previous naming scheme backwards compatible for add-ons.

I've asked for some help finishing up the documentation but this PR LGTM otherwise.

**Review by seanbudd — APPROVED** [review]

**Pull request merged**